pystac_monty/sources/ifrc_events.py

```diff
@@ -0,0 +1,163 @@
+    ifrc_impacts_collection_id = "ifrc-impacts"
+    ifrc_impacts_collection_url = ""  # TODO
+    hazard_profiles = HazardProfiles()
+
+    def __init__(self, data_source: IfrcEventsDataSource, geocoder: MontyGeoCoder):
+        self.data_source = data_source
+        self.geocoder = geocoder
+
+    def get_items(self):
+        print("\n", self.data_source.event_url)
+        return []
+
+    def get_event_collection(self) -> Collection:
+        """Get event collection"""
+        response = requests.get(self.ifrc_events_collection_url)
+        collection_dict = json.loads(response.text)
+        return Collection.from_dict(collection_dict)
+
+    def make_items(self) -> List[Item]:
+        """Create items"""
+        items = []
+
+        event_items = self.make_source_event_items()
+        items.extend(event_items)
+
+        # impact_items = self.make_impact_items()
+        # items.extend(impact_items)
+
+        return items
+
+    def make_source_event_items(self) -> List[Item]:
+        """Create ifrc event item"""
+        items = []
+        ifrc_data = self.check_and_get_ifrc_data()
+        if not ifrc_data:
+            return []
+
+        for data in ifrc_data:
+            item = self.make_source_event_item(data=data)
+            items.append(item)
+        return items
+
+    def make_source_event_item(self, data: dict) -> Item:
+        """Create ane event item"""
+        geometry = None
+        bbox = None
+        geom_data = self.geocoder.get_geometry_by_country_name(data["country"]["iso3"])
+
+        if geom_data:
+            geometry = geom_data["geometry"]
+            bbox = geom_data["bbox"]
+
+        # Create item
+        item = Item(
+            id=f"{STAC_EVENT_ID_PREFIX}{data["aid"]}",
+            geometry=geometry,
+            bbox=bbox,
+            datetime=data["start_date"],
+            start_datetime=data["start_date"],
+            end_datetime=data["end_date"],
+            properties={
+                    "title": data[""],
+            },
+        )
+
+        # Add Monty extension
+        MontyExtension.add_to(item)
+        monty = MontyExtension.ext(item)
+        monty.episode_number = 1  # IFRC DREF doesn't have episodes
+        monty.hazard_codes = self._map_ifrc_to_hazard_codes(data["dtype"]["name"])
+        monty.country_codes = data["country"]["iso3"]
+        monty.compute_and_set_correlation_id(hazard_profiles=self.hazard_profiles)
+
+        # Set collection and roles
+        item.set_collection(self.get_event_collection())
+        item.properties["roles"] = ["source", "event"]
+
+        # Filter out relevant disaster types
+        monty_accepted_disaster_types = {
+            "Earthquake", "Cyclone", "Volcanic Eruption", "Tsunami", "Flood",
+            "Cold Wave", "Fire", "Heat Wave", "Drought", "Storm Surge",
+            "Landslide", "Flash Flood"
+        }
+        if data["dtype"]["name"] not in monty_accepted_disaster_types:
+            return []
```

Comment:
We can filter these data in data validation in `self.check_and_get_ifrc_data()`. If you want to keep it here in this func, lets do it in the begining.